MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -38,211 +42,178 @@ Initialise the Interface
+* @param command : [input] The command (action) to add
 */
-void QtReflMainView::setTransferMethods(const std::set<std::string> &methods) {
-  for (auto method = methods.begin(); method != methods.end(); ++method) {
-    ui.comboTransferMethod->addItem((*method).c_str());
-  }
-}
+void QtReflMainView::addToMenu(QMenu *menu, ReflCommandBase_uptr command) {
 
-/**
-Set a new model in the tableview
-@param model : the model to be attached to the tableview
-*/
-void QtReflMainView::showTable(QReflTableModel_sptr model) {
-  m_model = model;
-  // So we can notify the presenter when the user updates the table
-  connect(m_model.get(),
-          SIGNAL(dataChanged(const QModelIndex &, const QModelIndex &)), this,
-          SLOT(tableUpdated(const QModelIndex &, const QModelIndex &)));
-  ui.viewTable->setModel(m_model.get());
-  ui.viewTable->resizeColumnsToContents();
-  std::string windowTitle = "ISIS Reflectometry (Polref) - " + m_toOpen;
-  auto mainWindowWidget = this->topLevelWidget();
-  mainWindowWidget->setWindowTitle(QString::fromStdString(windowTitle + "[*]"));
-  this->setWindowModified(false);
+  QAction *action = new QAction(QString::fromStdString(command->name()), this);
+  action->setIcon(QIcon(QString::fromStdString(command->icon())));
+  menu->addAction(action);
+  ReflCommandAdapter *adapter = new ReflCommandAdapter(std::move(command));
+  connect(action, SIGNAL(triggered()), adapter, SLOT(call()));
 }
 
 /**
-Set a new model for search results
-@param model : the model to be attached to the search results
+* Adds actions to the "Reflectometry" menu
 */
-void QtReflMainView::showSearch(ReflSearchModel_sptr model) {
-  m_searchModel = model;
-  ui.tableSearchResults->setModel(m_searchModel.get());
-  ui.tableSearchResults->resizeColumnsToContents();
-}
+void QtReflMainView::createReflectometryActions() {
 
-/**
-Set the list of tables the user is offered to open
-@param tables : the names of the tables in the ADS
-*/
-void QtReflMainView::setTableList(const std::set<std::string> &tables) {
-  ui.menuOpenTable->clear();
-  for (auto it = tables.begin(); it != tables.end(); ++it) {
-    QAction *openTable =
-        ui.menuOpenTable->addAction(QString::fromStdString(*it));
-    openTable->setIcon(QIcon("://worksheet.png"));
-
-    // Map this action to the table name
-    m_openMap->setMapping(openTable, QString::fromStdString(*it));
-    // When repeated corrections happen the QMessageBox from openTable()
-    // method in ReflMainViewPresenter will be called multiple times
-    // when 'no' is clicked.
-    // ConnectionType = UniqueConnection ensures that
-    // each object has only one of these signals.
-    connect(openTable, SIGNAL(triggered()), m_openMap, SLOT(map()),
-            Qt::UniqueConnection);
-    connect(m_openMap, SIGNAL(mapped(QString)), this, SLOT(setModel(QString)),
-            Qt::UniqueConnection);
-  }
-}
+  auto commands = ui.qReflTableView->getTablePresenter()->publishCommands();
```

Comment:
I think these commands should be pushed to the view. This view knows too much about the presenter structure and commands to call. `notify` is the only thing we want the view to call in the current setup.